src/orion/core/worker/strategies.py

```diff
@@ -0,0 +1,123 @@
+# -*- coding: utf-8 -*-
+"""
+:mod:`orion.core.worker.strategies` -- register objectives for incomplete trials
+========================================================================
+
+.. module:: strategies
+   :platform: Unix
+   :synopsis: Strategies to register objectives for incomplete trials.
+
+"""
+from abc import (ABCMeta, abstractmethod)
+import logging
+
+from orion.core.utils import Factory
+from orion.core.worker.trial import Trial
+
+log = logging.getLogger(__name__)
+
+
+def get_objective(trial):
+    """Get the value for the objective, if it exists, for this trial
+
+    :return: Float or None
+        The value of the objective, or None if it doesn't exist
+    """
+    objectives = [result['value'] for result in trial.results
+                  if result['type'] == 'objective']
+
+    if not objectives:
+        objective = None
+    elif len(objectives) == 1:
+        objective = objectives[0]
+    elif len(objectives) > 1:
+        raise RuntimeError("Trial %d has %d objectives", trial.id, len(objectives))
+
+    return objective
+
+
+class BaseParallelStrategy(object, metaclass=ABCMeta):
```

Comment:
There must be a factory class which inherits from BaseParallelStrategy and has metaclass=Factory in order to support creation of parallel strategies using the meta class You can look at OptimizationAlgorithm for a inspiration. Also, I would define parallel strategies outside the core, just like algos, so that users can customize them as well. Parallel strategies are heuristics and there is world of possibilities to explore.